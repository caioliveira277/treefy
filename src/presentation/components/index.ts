export * from './button/button-component';
export * from './text-input/text-input-component';
export * from './link/link-component';
export * from './legend/legend-component';
export * from './picker-input/picker-input-component';
export * from './page-loading/page-loading-component';
export * from './toast-message/toast-message-component';
export * from './skeleton-loading/skeleton-loading-component';
export * from './empty-content/empty-content-component';
export * from './search-input/search-input-component';
export * from './animated-height/animated-height';
export * from './loading-overlay/loading-overlay-component';
