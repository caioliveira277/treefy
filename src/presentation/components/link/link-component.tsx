import { StyleProp, ViewStyle } from 'react-native';
import { useTheme } from 'styled-components';
import { Container, Link, Text } from './styles';

export interface LinkComponentProps {
  children: string;
  style?: StyleProp<ViewStyle>;
  onPress?: () => void;
  fontSize?: string;
  color?: string;
}

export const LinkComponent: React.FC<LinkComponentProps> = ({
  children,
  style,
  onPress,
  fontSize,
  color,
}) => {
  const theme = useTheme();
  return (
    <Container testID="container" style={style}>
      <Link testID="link" onPress={onPress}>
        <Text testID="text" theme={theme} fontSize={fontSize} color={color}>
          {children}
        </Text>
      </Link>
    </Container>
  );
};
