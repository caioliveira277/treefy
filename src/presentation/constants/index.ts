export const TREEFY_SUPPORT = {
  email: 'treefy.suporte@gmail.com',
  cel1: '(15) 99782-7880',
  cel2: '(11) 93203-7301',
};
