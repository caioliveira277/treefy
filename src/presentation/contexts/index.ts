export * from './authentication-context';
export * from './introduction-context';
export * from './toast-context';
