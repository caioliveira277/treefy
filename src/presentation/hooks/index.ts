export * from './use-debounce/use-debounce-hook';
