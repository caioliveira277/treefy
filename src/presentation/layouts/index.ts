export * from './public/public-layout';
export * from './profile/profile-layout';
