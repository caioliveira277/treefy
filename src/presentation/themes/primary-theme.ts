import { DefaultTheme } from 'styled-components';

export const primaryTheme: DefaultTheme = {
  colors: {
    primary: '#4ABC86',
    secondary: '#253239',
    body: '#585960',
    white: '#ffffff',
    contrast: '#FCFDFF',
    placeholder: '#C6C6C6',
    placeholder_light: '#EEEEEE',
    rate_active: '#D56D6D',
    rate_inactive: '#E5A6A7',
    sun: '#FEF3A7',
    water: '#B0D8FE',
    edit: '#4ABCBC',
    complete: '#4ABC86',
  },
  borders: {
    border_radius_sm: '5px',
    border_radius_md: '10px',
  },
  fonts: {
    sizes: {
      xl: '18px',
      lg: '16px',
      md: '14px',
      sm: '12px',
    },
    families: {
      regular: 'RobotoSlab_400Regular',
      medium: 'RobotoSlab_500Medium',
      bold: 'RobotoSlab_700Bold',
    },
  },
  shadows: {
    sm: {
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.23,
      shadowRadius: 2.62,
      elevation: 4,
    },
  },
};

// shadow_sm: '0px 2px 4px rgba(0, 0, 0, 0.1)',
