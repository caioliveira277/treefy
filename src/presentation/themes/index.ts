import { primaryTheme } from './primary-theme';
export const currentTheme = (() => primaryTheme)();
