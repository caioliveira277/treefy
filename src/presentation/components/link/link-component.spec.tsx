import { mockRender } from '@/presentation/test';
import { LinkComponent, LinkComponentProps } from './link-component';
import faker from '@faker-js/faker';
import { fireEvent, RenderAPI } from '@testing-library/react-native';
import { currentTheme } from '@/presentation/themes';
import 'jest-styled-components/native';

type SutTypes = {
  sut: RenderAPI;
};
interface MakeSutProps extends Partial<LinkComponentProps> {}

const makeSut = ({ children = 'anyText', onPress }: MakeSutProps): SutTypes => {
  const sut = mockRender(
    <LinkComponent onPress={onPress}>{children}</LinkComponent>
  );
  return {
    sut,
  };
};

describe('LinkComponent', () => {
  test('Should match with a snapshot', () => {
    const { sut } = makeSut({});
    const { toJSON } = sut;
    expect(toJSON()).toMatchSnapshot();
  });
  test('Should display the text passed by children', () => {
    const anyText = faker.datatype.string();
    const { sut } = makeSut({
      children: anyText,
    });
    const { getByTestId } = sut;
    const text = getByTestId('text');
    expect(text.children).toContain(anyText);
  });
  test('Should trigger the press event', async () => {
    const handlePress = jest.fn();
    const { sut } = makeSut({
      onPress: handlePress,
    });
    const { getByTestId } = sut;
    const link = getByTestId('link');
    fireEvent.press(link);
    expect(handlePress).toHaveBeenCalledTimes(1);
  });
  test('Should apply the primary color in the text', () => {
    const { sut } = makeSut({});
    const { getByTestId } = sut;
    const text = getByTestId('text');

    expect(text).toHaveStyleRule('color', currentTheme.colors.primary);
  });
});
