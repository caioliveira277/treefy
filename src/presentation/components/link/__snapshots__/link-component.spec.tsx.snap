// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkComponent Should match with a snapshot 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
  testID="container"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="link"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4ABC86",
            "fontFamily": "RobotoSlab_400Regular",
            "fontSize": 14,
            "textDecorationColor": "black",
            "textDecorationLine": "underline",
            "textDecorationStyle": "solid",
          },
        ]
      }
      testID="text"
      theme={
        Object {
          "borders": Object {
            "border_radius_md": "10px",
            "border_radius_sm": "5px",
          },
          "colors": Object {
            "body": "#585960",
            "complete": "#4ABC86",
            "contrast": "#FCFDFF",
            "danger": "#E04244",
            "edit": "#4ABCBC",
            "placeholder": "#C6C6C6",
            "placeholder_light": "#EEEEEE",
            "primary": "#4ABC86",
            "rate_active": "#D56D6D",
            "rate_inactive": "#E5A6A7",
            "secondary": "#253239",
            "sun": "#FEF3A7",
            "water": "#B0D8FE",
            "white": "#ffffff",
          },
          "fonts": Object {
            "families": Object {
              "bold": "RobotoSlab_700Bold",
              "medium": "RobotoSlab_500Medium",
              "regular": "RobotoSlab_400Regular",
            },
            "sizes": Object {
              "lg": "16px",
              "md": "14px",
              "sm": "12px",
              "xl": "18px",
            },
          },
          "shadows": Object {
            "sm": Object {
              "elevation": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2.62,
            },
          },
        }
      }
    >
      anyText
    </Text>
  </View>
</View>
`;
